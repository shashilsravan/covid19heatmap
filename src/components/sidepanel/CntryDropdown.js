import React from 'react'
import Option from '../../minicomponents/Option'

export default function CntryDropdown({handleChange, countryName, country}) {
    return (
        <select className="countrySelect" name="country" 
            onChange={handleChange} value={countryName} required>
                <option> Select a Country </option>
                {country && country.map(cntry => 
                    <Option key={cntry.ID} data={cntry} />
                )}
        </select>
    )
}
