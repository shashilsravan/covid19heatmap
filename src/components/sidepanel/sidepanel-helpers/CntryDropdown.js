import React from 'react'
import Option from '../../../globalComponents/Option'

export default function CntryDropdown({handleChange, countryName, country}) {
    return (
        <select className="countrySelect" name="country" 
            onChange={handleChange} value={countryName} required>
                <option> Select a Country </option>
                {country && country.map(cntry => 
                    <Option key={cntry.ID} data={cntry} />
                )}
        </select>
    )
}
