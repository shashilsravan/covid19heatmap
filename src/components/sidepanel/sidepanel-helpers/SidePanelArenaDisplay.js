import React from 'react'
import CountryImage from '../../../globalComponents/CountryImage'

export default function SidePanelArenaDisplay({data}) {
    const tableStyling = {
        display: "flex", alignItems: "center"
    }
    const {Country, CountryCode, TotalConfirmed} = data
    return (
        <tr>
            <td style={tableStyling}>
                <CountryImage CountryCode={CountryCode} />
                {Country.length < 12 ? Country : `${Country.substring(0, 11)} ...`}
            </td>
            <td>
                {TotalConfirmed}
            </td>
        </tr>
    )
}
