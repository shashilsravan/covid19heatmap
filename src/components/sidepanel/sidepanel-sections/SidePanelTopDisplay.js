import React from 'react'
import Arena from './Arena'
import Global from './Global'
import Button from '../../../globalComponents/Button'

export default function SidePanelTopDisplay({arena, handleGlobal, handleCountry, global, country}) {
    return (
        <div className="displayArea">
            <div className="buttons">
                <Button styling={arena == 'global' ? {background: "black", color: "white"} : {background: "white", color: "black"}}
                    change={handleGlobal} text="Global" />
                <Button styling={arena == 'country' ? {background: "black", color: "white"} : {background: "white", color: "black"}}
                    change={handleCountry} text="Country" />
            </div>
            {arena == 'global' && <Global global={global} />}
            {arena == 'country' && <Arena country={country} />}
        </div>
    )
}
