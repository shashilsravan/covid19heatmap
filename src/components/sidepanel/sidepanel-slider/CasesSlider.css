.my-slider {
  padding: 4px;
}
.confirmed {
  font-size: 0.7rem;
  background: linear-gradient(to bottom, #32de84, #ace1af);
  padding-bottom: 8px;
  box-shadow: inset 0 0 6px rgba(255, 255, 255, 1);
}
.confirmed h4 {
  font-style: italic;
  text-transform: uppercase;
  margin: 0;
  padding: 6px 0;
}
.confirmed p {
  margin: 0;
  font-weight: bold;
  font-size: 0.85rem;
  padding-bottom: 6px;
}

.active {
  font-size: 0.7rem;
  background: linear-gradient(to bottom, #007fff, #89cff0);
  padding-bottom: 8px;
  box-shadow: inset 0 0 6px rgba(255, 255, 255, 1);
}
.active h4 {
  font-style: italic;
  text-transform: uppercase;
  margin: 0;
  padding: 6px 0;
}
.active p {
  margin: 0;
  font-weight: bold;
  font-size: 0.85rem;
  padding-bottom: 6px;
}

.deaths {
  font-size: 0.7rem;
  background: linear-gradient(to bottom, #aa0000, #fd5c63);
  padding-bottom: 8px;
  box-shadow: inset 0 0 6px rgba(255, 255, 255, 1);
}
.deaths h4 {
  font-style: italic;
  text-transform: uppercase;
  margin: 0;
  padding: 6px 0;
}
.deaths p {
  margin: 0;
  font-weight: bold;
  font-size: 0.85rem;
  padding-bottom: 6px;
}
