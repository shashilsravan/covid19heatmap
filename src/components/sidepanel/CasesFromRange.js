import React from 'react'
import DateInput from '../../minicomponents/DateInput'
import Button from '../../minicomponents/Button'
import CntryDropdown from './CntryDropdown'

export default function CasesFromRange({handleChange, selectionCntry, country, 
startDate, endDate, handleClick}) {
    return (
        <div className="top-area">
            <CntryDropdown handleChange={handleChange} countryName={selectionCntry}
                country={country} />
            <div className="dates">
                <DateInput startDate={startDate} value={startDate}
                    change={handleChange} name="startDate" />
                <DateInput startDate={startDate} value={endDate}
                    change={handleChange} name="endDate" />
            </div>
            <Button text="Get Data" change={handleClick} nameOfClass="btn" />
        </div>
    )
}
