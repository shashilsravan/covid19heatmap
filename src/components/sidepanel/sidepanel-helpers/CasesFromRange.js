import React from 'react'
import DateInput from '../../../globalComponents/DateInput'
import Button from '../../../globalComponents/Button'
import CntryDropdown from './CntryDropdown'

export default function CasesFromRange({handleChange, selectionCntry, country, 
startDate, endDate, handleClick}) {
    return (
        <div className="top-area">
            <CntryDropdown handleChange={handleChange} countryName={selectionCntry}
                country={country} />
            <div className="dates">
                <DateInput value={startDate} change={handleChange} name="startDate" />
                <DateInput value={endDate} change={handleChange} name="endDate" />
            </div>
            <Button text="Get Data" change={handleClick} nameOfClass="btn" />
        </div>
    )
}
