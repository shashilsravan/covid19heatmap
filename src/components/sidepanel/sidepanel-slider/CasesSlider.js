import React, { useEffect, useState } from 'react'
import './CasesSlider.css'
import { getRangeCases } from '../sidepanel-helpers/SidePanelData'
import EachSlide from './EachSlide';
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import Slider from "react-slick";

export default function CasesSlider({country, startDate, endDate}) {
    const [confirmed, setConfirmed] = useState({
        current: 0, old: 0, new: 0
    })
    const [active, setActive] = useState({
        current: 0, old: 0, new: 0
    })
    const [deaths, setDeaths] = useState({
        current: 0, old: 0, new: 0
    })

    const getData = async () => {
        const res = await getRangeCases(country, startDate, endDate)
        const temp1 = res[0][res[0].length - 1]
        const temp2 = res[0][0]

        setConfirmed({
            current: temp1.Confirmed - temp2.Confirmed, 
            old: temp2.Confirmed, new: temp1.Confirmed
        })

        setActive({
            current: temp1.Active - temp2.Active,
            old: temp2.Active, new: temp1.Active
        })

        setDeaths({
            current: temp1.Deaths - temp2.Deaths,
            old: temp1.Deaths, new: temp2.Deaths
        })
    };

    const settings = {
        infinite: true, autoplay: true,
        autoplaySpeed: 4000, arrows: false, speed: 500,
        slidesToShow: 1.1, slidesToScroll: 1
    };

    useEffect(() => {
        getData()
    }, [])
    return (
        <div className="my-slider">
            <Slider {...settings}>
                <EachSlide styling="confirmed" title="Confirmed cases:"
                cases={confirmed.current} oldCases={confirmed.old} newCases={confirmed.new} />
                
                <EachSlide styling="active" title="Active cases:" 
                cases={active.current} oldCases={active.old} newCases={active.new} />
                
                <EachSlide styling="deaths" title="Deaths:" cases={deaths.current}
                oldCases={deaths.old} newCases={deaths.new} />
            </Slider>
        </div>
    )
}
