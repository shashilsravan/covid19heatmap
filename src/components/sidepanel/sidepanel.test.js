import React from "react"
import Enzyme, { shallow, mount, render } from "enzyme"
import EachSlide from "./sidepanel-slider/EachSlide";
import CasesSlider from "./sidepanel-slider/CasesSlider";
import CasesFromRange from "./sidepanel-helpers/CasesFromRange";
import toJson from 'enzyme-to-json';
import CntryDropdown from "./sidepanel-helpers/CntryDropdown";
import SidePanelArenaDisplay from "./sidepanel-helpers/SidePanelArenaDisplay";
import SidePanelTopDisplay from "./sidepanel-sections/SidePanelTopDisplay";

const eachSlideData = {
    title: "Sample title",
    oldCases: 100,
    cases: 200,
    newCases: 150
}

describe("Slider of Sidepanel", () => {
    it("Each Slide Title Check", () => {
        const eachSlide = mount(<EachSlide title={eachSlideData.title} />)
        const title = eachSlide.find("h4").text()
        expect(title).toEqual(eachSlideData.title)
    });
    it("Each Slide Cases Check", () => {
        const eachSlide = mount(<EachSlide cases={eachSlideData.cases} />)
        const cases = eachSlide.find("p").text().split()[0]
        expect(parseInt(cases)).toEqual(eachSlideData.cases)
    });
    it("Overall Slider check", () => {
        shallow(<CasesSlider />)
    })
})

describe("", () => {
    it("Cases From Range check", () => {
        shallow(<CasesFromRange />)
    })
    it("Country dropdown check", () => {
        shallow(<CntryDropdown />)
    })
    // snapshot testing
    it("Sidepanel Arena display check", () => {
        const wrapper = shallow(<SidePanelArenaDisplay 
            data={{Country: "India", CountryCode: "In", TotalConfirmed: 3000}} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})