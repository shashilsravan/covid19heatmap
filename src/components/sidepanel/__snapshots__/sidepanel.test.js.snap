// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Sidepanel Arena display check 1`] = `
<tr>
  <td
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
      }
    }
  >
    <CountryImage
      CountryCode="In"
    />
    India
  </td>
  <td>
    3000
  </td>
</tr>
`;
